src/test/ui/pattern/usefulness/deny-irrefutable-let-patterns.stderr

```diff
@@ -1,8 +1,8 @@
 error: irrefutable `if let` pattern
-  --> $DIR/deny-irrefutable-let-patterns.rs:7:5
+  --> $DIR/deny-irrefutable-let-patterns.rs:7:12
    |
 LL |     if let _ = 5 {}
-   |     ^^^^^^^^^^^^^^^
+   |            ^
```

Comment:
The "irrefutable pattern" lint changed its span from the whole if-let/while-let expression to span of the specific pattern which is irrefutable.
Looks like an acceptable and maybe desirable change.

The "macro" tests (`if-let.rs` and `while-let.rs`) also changed for exactly the same reason.

Reply:
Maybe add the whole `pattern = expression` span for `hir::ExprKind::Let`?

```
Let {
  expr: Expr,
  pat: Pat,
  span: Span
}
```

Reply:
Maybe, if you really want to preserve the old behavior.

Reply:
My 2c: New, small span is better.